### PR TITLE
fix: input-number in step-strictly onChange bug

### PR DESCRIPTION
**Pull request by chenxch**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by chenxch** [comment]
fix #6659

File: packages/components/input-number/src/input-number.vue [hunk 1]
```diff
@@ -193,6 +193,13 @@ export default defineComponent({
     }
     const setCurrentValue = (newVal: number | string) => {
       const oldVal = data.currentValue
+      if (props.stepStrictly) {
+        const stepPrecision = getPrecision(props.step)
+        const precisionFactor = 10 ** stepPrecision
+        newVal =
+          (Math.round(newVal / props.step) * precisionFactor * props.step) /
+          precisionFactor
+      }
```
Comment (JeremyWuuuuu) [inline packages/components/input-number/src/input-number.vue:202]: @chenxch You did an excellent job, but the code seems duplicated with: [_decrease](https://github.com/element-plus/element-plus/blob/adf1ecf3ebffaf92dc59119fcb0015d8f1607ffe/packages/components/input-number/src/input-number.vue#L173), would you mind encapsulating them?

Comment (JeremyWuuuuu) [reply packages/components/input-number/src/input-number.vue:202]: @chenxch Nice! This is a good start for duplication reduction. Thank you for your cooperation! :bow:

Comment (JeremyWuuuuu) [reply packages/components/input-number/src/input-number.vue:202]: We can address this in an additional PR, I'll merge this first to fix the bug.

**Review by jw-foss — APPROVED** [review]

**Pull request merged**